backends/avx/ceed-avx-tensor.c

```diff
@@ -97,26 +114,26 @@ static inline int CeedTensorContract_Avx_Remainder(CeedTensorContract contract,
     for (CeedInt c = (C / CC) * CC; c < C; c += 4) {
       // Blocks of 4 rows
       for (CeedInt j = 0; j < J_break; j += JJ) {
-        __m256d vv[JJ];  // Output tile to be held in registers
-        for (CeedInt jj = 0; jj < JJ; jj++) vv[jj] = _mm256_loadu_pd(&v[(a * J + j + jj) * C + c]);
+        rtype vv[JJ];  // Output tile to be held in registers
+        for (CeedInt jj = 0; jj < JJ; jj++) vv[jj] = loadu(&v[(a * J + j + jj) * C + c]);
 
         for (CeedInt b = 0; b < B; b++) {
-          __m256d tqu;
-          if (C - c == 1) tqu = _mm256_set_pd(0.0, 0.0, 0.0, u[(a * B + b) * C + c + 0]);
-          else if (C - c == 2) tqu = _mm256_set_pd(0.0, 0.0, u[(a * B + b) * C + c + 1], u[(a * B + b) * C + c + 0]);
-          else if (C - c == 3) tqu = _mm256_set_pd(0.0, u[(a * B + b) * C + c + 2], u[(a * B + b) * C + c + 1], u[(a * B + b) * C + c + 0]);
-          else tqu = _mm256_loadu_pd(&u[(a * B + b) * C + c]);
+          rtype tqu;
+          if (C - c == 1) tqu = set(0.0, 0.0, 0.0, u[(a * B + b) * C + c + 0]);
```

Comment:
ah, I think this is why I didn't use 8 floats originally

Reply:
There's probably a way around this, but I haven't invested time since floats aren't a hotspot for performance in any of our current applications.

Reply:
Yeah the block size of 4 is pretty hardcoded throughout, I would shy away from changing it if it's not an issue.